packages/tools/kadena-cli/src/devnet/utils/simulation/file.ts

```diff
@@ -0,0 +1,55 @@
+import fs from 'fs';
+import path from 'path';
+
+export interface IFileData {
+  timestamp: number;
+  from: string;
+  to: string;
+  amount: number;
+  requestKey: string;
+  action: TransferType | TokenActionType;
+}
+
+export interface IFileError {
+  error: string;
+}
+
+export type TransferType =
+  | 'fund'
+  | 'transfer'
+  | 'cross-chain-transfer'
+  | 'safe-transfer'
+  | undefined;
+
+export type TokenActionType = 'create' | 'mint' | 'transfer' | undefined;
+
+export function createDir(directory: string): void {
+  if (!fs.existsSync(directory)) {
+    fs.mkdirSync(directory);
+  }
+}
+
+export function createFile(folder: string, filename: string): string {
+  const exampleData: IFileData = {
+    timestamp: 0,
+    from: '',
+    to: '',
+    amount: 0,
+    requestKey: '',
+    action: undefined,
+  };
+  const directory = path.join(`${folder}/`);
+  createDir(directory);
+  const filepath = path.join(directory, filename);
+  const headers = `${Object.keys(exampleData).join(',')}\n`;
+  fs.writeFileSync(filepath, headers);
+  return filepath;
+}
+
+export function appendToFile(
```

Comment:
@barthuijgen throughout this whole file helper you want it to make use of your service so it can use memoryfs?

Reply:
Yes definitely! This does seem like code that would be good to add unit test for. 

Please make use of the `services.filesystem` implementation, it does not have appendFile yet, but you can add it to the service, send me a message if it is unclear how to do that.

Reply:
Fixed